src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/CacheTokenOnStartupBackgroundService.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace Workleap.Extensions.Http.Authentication.ClientCredentialsGrant;
+
+internal sealed class CacheTokenOnStartupBackgroundService : BackgroundService
+{
+    private readonly IHostApplicationLifetime _applicationLifetime;
+    private readonly IClientCredentialsTokenManagementService _tokenManagementService;
+    private readonly IOptions<CacheTokenOnStartupBackgroundServiceOptions> _backgroundServiceOptions;
+    private readonly SemaphoreSlim _allTokensCachedSignal;
+    private readonly List<string> _clientNames;
+
+    private CancellationTokenRegistration? _applicationStartedRegistration;
+    private int _successfulCachedTokenCount;
+
+    public CacheTokenOnStartupBackgroundService(
+        IHostApplicationLifetime applicationLifetime,
+        IClientCredentialsTokenManagementService tokenManagementService,
+        IOptions<CacheTokenOnStartupBackgroundServiceOptions> backgroundServiceOptions)
+    {
+        this._applicationLifetime = applicationLifetime;
+        this._tokenManagementService = tokenManagementService;
+        this._backgroundServiceOptions = backgroundServiceOptions;
+        this._allTokensCachedSignal = new SemaphoreSlim(initialCount: 0, maxCount: 1);
+        this._clientNames = new List<string>();
+    }
+
+    protected override Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        this._clientNames.AddRange(this._backgroundServiceOptions.Value.ClientCredentialPoweredClientNames);
+
+        this._applicationStartedRegistration = this._applicationLifetime.ApplicationStarted.Register(() =>
+        {
+            foreach (var clientName in this._clientNames)
+            {
+                // We don't await this task because we want to cache all tokens in parallel (fire and forget)
+                _ = this.CacheTokenAsync(clientName, stoppingToken);
```

Comment:
You should do an actual fire and forget. If there is an error, it would report an TaskUnhandledException. 

https://www.meziantou.net/fire-and-forget-a-task-in-dotnet.htm